src/ShallowTraversal.js

```diff
@@ -110,7 +110,7 @@ export function buildPredicate(selector) {
           return node => nodeHasId(node, selector.slice(1));
 
         case SELECTOR.PROP_TYPE: {
-          const propKey = selector.split(/\[([a-zA-Z-]*?)(=|])/)[1];
+          const propKey = selector.split(/\[([a-zA-Z-0-9-]*?)(=|])/)[1];
```

Comment:
I definitely expect ShallowTraversal to throw an error - if you try to use an invalid selector, we should blow up instead of silently giving you a result.